src/circuit/gadget/ecc/chip/mul_fixed.rs

```diff
@@ -531,14 +531,14 @@ impl ScalarFixed {
     }
 
     // The scalar decomposition is guaranteed to be in three-bit windows,
-    // so we also cast the least significant byte in their serialisation
+    // so we also cast the least significant 4 bytes in their serialisation
     // into usize for convenient indexing into `u`-values
     fn windows_usize(&self) -> Vec<Option<usize>> {
         self.windows_field()
             .iter()
             .map(|window| {
                 if let Some(window) = window {
-                    let window = window.to_bytes()[0] as usize;
+                    let window = window.get_lower_32() as usize;
```

Comment:
@daira left a comment inside the `get_lower_32()` impls:
> // TODO: don't reduce, just hash the Montgomery form. (Requires rebuilding perfect hash table.)

We'll need to take care over there to ensure we don't alter this API, now that we are relying on it outside the perfect hash table.